### ISSUE TITLE
InvalidCountException even though the mock method is called

### ISSUE THREAD

**Issue by maxbethke** [padraic/mockery#1352]
### Mockery Version

1.6.7

### PHP Version

PHP 8.3

### Issue Description

When testing certain method calls on a constructor-injected mock, Mockery is throwing a `Mockery\Exception\InvalidCountException`, indicating that the method wasn't called, even though it is called in the tested code. The specific method in question is `fromYaml()` from the `League\OpenAPIValidation\PSR7\ValidatorBuilder` class.

### Steps to Reproduce

The following code can be used to reproduce the issue:
```
<?php

require 'vendor/autoload.php';

use League\OpenAPIValidation\PSR7\ValidatorBuilder;
use Mockery\Adapter\Phpunit\MockeryTestCase;

class SchemaValidatorService
{
    public function __construct(private ValidatorBuilder $validatorBuilder)
    {
    }

    public function validateRoutedRequest()
    {
        return $this->validatorBuilder->fromYaml('testYaml')->getRoutedRequestValidator();
    }
}

class SchemaValidatorServiceTest extends MockeryTestCase
{
    private $schemaValidatorService;
    private $validatorBuilder;

    protected function setUp(): void
    {
        $this->validatorBuilder = \Mockery::mock(ValidatorBuilder::class);
        $this->validatorBuilder->shouldReceive('fromYaml')->andReturnSelf();
        $this->schemaValidatorService = new SchemaValidatorService($this->validatorBuilder);
    }

    public function testThatItUsesTheApiSpecification()
    {
        $this->validatorBuilder->shouldReceive('fromYaml')->once()->andReturnSelf();
        $this->schemaValidatorService->validateRoutedRequest();
    }

    protected function tearDown(): void
    {
        \Mockery::close();
    }
}

```

### Expected Behavior

The test should pass

### Actual Behavior

`Mockery\Exception\InvalidCountException : Method fromYaml(<Any Arguments>) from Mockery_1_League_OpenAPIValidation_PSR7_ValidatorBuilder should be called
 exactly 1 times but called 0 times.` is thown and the test fails

### Exception or Error

Mockery\Exception\InvalidCountException

### Additional Information

The test passes, when I use PHPUnits `createMock` instead of mockery like this
```
    protected function setUp(): void
    {
        $this->validatorBuilder = $this->createMock(ValidatorBuilder::class);
        $this->validatorBuilder->expects($this->once())
                               ->method('fromYaml')
                               ->willReturn($this->validatorBuilder);
        $this->schemaValidatorService = new SchemaValidatorService($this->validatorBuilder);
    }
```

**Comment by ghostwriter** [comment]
Hey @maxbethke,

Thanks for reporting your findings.

> ### Issue Description
> When testing certain method calls on a constructor-injected mock, 
> Mockery is throwing a `Mockery\Exception\InvalidCountException`,
> indicating that the method wasn't called, even though it is called in the tested code.
>
> The specific method in question is `fromYaml()` from the `League\OpenAPIValidation\PSR7\ValidatorBuilder` class.
<hr>

It looks like the mock object is properly configured to expect the `fromYaml` method call `2` times.

- once inside `setUp`

- once inside `testThatItUsesTheApiSpecification`

Each time the `shouldReceive` method is called on a mock object in Mockery,
it adds a new expectation for that method with the specified arguments.

<hr>

> The test passes, when I use PHPUnits `createMock` instead of mockery like this
> 
> ```
>     protected function setUp(): void
>     {
>         $this->validatorBuilder = $this->createMock(ValidatorBuilder::class);
>         $this->validatorBuilder->expects($this->once())
>                                ->method('fromYaml')
>                                ->willReturn($this->validatorBuilder);
>         $this->schemaValidatorService = new SchemaValidatorService($this->validatorBuilder);
>     }
> ```

The equivalent Mockery code for the above code would look like this:

``` diff
class SchemaValidatorServiceTest extends MockeryTestCase
{
    private $schemaValidatorService;
    private $validatorBuilder;

    protected function setUp(): void
    {
        $this->validatorBuilder = \Mockery::mock(ValidatorBuilder::class);
+        $this->validatorBuilder->shouldReceive('fromYaml')->once()->andReturnSelf();
        $this->schemaValidatorService = new SchemaValidatorService($this->validatorBuilder);
    }

    public function testThatItUsesTheApiSpecification()
    {
-        $this->validatorBuilder->shouldReceive('fromYaml')->once()->andReturnSelf();
        $this->schemaValidatorService->validateRoutedRequest();
    }

    protected function tearDown(): void
    {
        \Mockery::close();
    }
}
```

To summarize, the code you provided sets up 2 expectations for the mock object.

once inside `setUp` and another in `testThatItUsesTheApiSpecification`

The test fails because that method is called exactly 1 time in `testThatItUsesTheApiSpecification`
but expected to be called 2 times.

This is the expected behavior.

Let me know if this resolves the issue for you. (please don't forget to closing this issue.)
